src/payment/payment-strategy-action-creator.ts

```diff
@@ -47,8 +47,17 @@ export default class PaymentStrategyActionCreator {
                     strategy = this._strategyRegistry.get(PaymentStrategyType.NO_PAYMENT_DATA_REQUIRED);
                 }
 
-                return strategy
-                    .execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId })
+                return strategy.execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId })
+                    .catch(error => {
+                        return getGoogleRecapthaToken(store, error)
+                            .then((token: string) => {
+                                if (payload.payment) {
+                                    payload.payment.paymentRecaptchaToken = token;
```

Comment:
Can we create a new object instead here instead of modifying the argument by reference? This is because the `payload` is passed in from the SDK client and we don't want to modify it intentionally.